// @clr/angular/esm2020/forms/common/layout.mjs
/*
 * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./providers/layout.service";
export class ClrLayout {
    constructor(layoutService) {
        this.layoutService = layoutService;
    }
    ngOnInit() {
        // Only set the layout if it is a valid option
        if (this.layout && this.layoutService.isValid(this.layout)) {
            this.layoutService.layout = this.layout;
        }
    }
}
ClrLayout.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: ClrLayout, deps: [{ token: i1.LayoutService }], target: i0.ɵɵFactoryTarget.Directive });
ClrLayout.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.3.11", type: ClrLayout, selector: "[clrForm][clrLayout]", inputs: { layout: ["clrLayout", "layout"] }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: ClrLayout, decorators: [{
            type: Directive,
            args: [{
                    selector: '[clrForm][clrLayout]',
                }]
        }], ctorParameters: function () { return [{ type: i1.LayoutService }]; }, propDecorators: { layout: [{
                type: Input,
                args: ['clrLayout']
            }] } });
