// @clr/angular/esm2020/forms/common/providers/ng-control.service.mjs
/*
 * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
export class NgControlService {
    constructor() {
        // Observable to subscribe to the control, since its not available immediately for projected content
        this._controlChanges = new Subject();
        this._helpers = new Subject();
    }
    get controlChanges() {
        return this._controlChanges.asObservable();
    }
    setControl(control) {
        this._controlChanges.next(control);
    }
    get helpersChange() {
        return this._helpers.asObservable();
    }
    setHelpers(state) {
        this._helpers.next(state);
    }
}
NgControlService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: NgControlService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
NgControlService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: NgControlService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: NgControlService, decorators: [{
            type: Injectable
        }] });
