// @clr/angular/esm2020/emphasis/alert/providers/multi-alert.service.mjs
/*
 * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
export class MultiAlertService {
    constructor() {
        /**
         * The Observable that lets other classes subscribe to changes
         */
        this._change = new Subject();
    }
    get changes() {
        return this._change.asObservable();
    }
    get current() {
        return this._current;
    }
    set current(index) {
        if (index !== this._current) {
            this._current = index;
            this._change.next(index);
        }
    }
    get activeAlerts() {
        return this.allAlerts && this.allAlerts.filter(alert => !alert._closed);
    }
    get currentAlert() {
        return this.activeAlerts && this.activeAlerts[this.current];
    }
    set currentAlert(alert) {
        this.current = this.activeAlerts.indexOf(alert);
    }
    get count() {
        return (this.activeAlerts && this.activeAlerts.length) || 0;
    }
    manage(alerts) {
        if (this.subscription) {
            this.subscription.unsubscribe();
        }
        this.allAlerts = alerts;
        // After receiving alerts' QueryList,
        // we are picking index 0 as current by default if a user hasn't any index
        this.current = typeof this._current === 'number' ? this._current : 0;
        // we have to also broadcast that initial index
        this._change.next(this.current);
        this.subscription = this.allAlerts.changes.subscribe(() => {
            if (this.current >= this.allAlerts.length) {
                this.current = Math.max(0, this.allAlerts.length - 1);
            }
        });
    }
    next() {
        this._current = this.current === this.activeAlerts.length - 1 ? 0 : this.current + 1;
        this._change.next(this._current);
    }
    previous() {
        if (this.activeAlerts.length === 0) {
            return;
        }
        this._current = this.current === 0 ? this.activeAlerts.length - 1 : this.current - 1;
        this._change.next(this._current);
    }
    open() {
        if (this.activeAlerts.length === 0) {
            return;
        }
        if (!this.currentAlert) {
            this._current = 0;
        }
        this._change.next(this._current);
    }
    close(isCurrentAlert) {
        if (this.activeAlerts.length === 0) {
            return;
        }
        if (isCurrentAlert) {
            this._current = Math.max(0, this.current - 1);
        }
        this._change.next(this._current);
    }
    destroy() {
        if (this.subscription) {
            this.subscription.unsubscribe();
        }
    }
}
MultiAlertService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: MultiAlertService, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
MultiAlertService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: MultiAlertService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: MultiAlertService, decorators: [{
            type: Injectable
        }] });
