// @clr/angular/esm2020/emphasis/alert/alert-item.mjs
/*
 * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Component } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./providers/icon-and-types.service";
import * as i2 from "../../icon/icon";
export class ClrAlertItem {
    constructor(iconService) {
        this.iconService = iconService;
    }
}
ClrAlertItem.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: ClrAlertItem, deps: [{ token: i1.AlertIconAndTypesService }], target: i0.ɵɵFactoryTarget.Component });
ClrAlertItem.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.3.11", type: ClrAlertItem, selector: "clr-alert-item", host: { classAttribute: "alert-item" }, ngImport: i0, template: `
    <div class="alert-icon-wrapper">
      <cds-icon
        class="alert-icon"
        role="img"
        [attr.shape]="iconService.alertIconShape"
        [attr.aria-label]="iconService.alertIconTitle"
      ></cds-icon>
    </div>
    <ng-content></ng-content>
  `, isInline: true, directives: [{ type: i2.CdsIconCustomTag, selector: "cds-icon" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: ClrAlertItem, decorators: [{
            type: Component,
            args: [{
                    selector: 'clr-alert-item',
                    template: `
    <div class="alert-icon-wrapper">
      <cds-icon
        class="alert-icon"
        role="img"
        [attr.shape]="iconService.alertIconShape"
        [attr.aria-label]="iconService.alertIconTitle"
      ></cds-icon>
    </div>
    <ng-content></ng-content>
  `,
                    host: { class: 'alert-item' },
                }]
        }], ctorParameters: function () { return [{ type: i1.AlertIconAndTypesService }]; } });
