// @clr/angular/esm2020/forms/common/if-control-state/if-control-state.service.mjs
/*
 * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import * as i0 from "@angular/core";
import * as i1 from "../providers/ng-control.service";
export var CONTROL_STATE;
(function (CONTROL_STATE) {
    CONTROL_STATE["NONE"] = "NONE";
    CONTROL_STATE["VALID"] = "VALID";
    CONTROL_STATE["INVALID"] = "INVALID";
})(CONTROL_STATE || (CONTROL_STATE = {}));
export class IfControlStateService {
    constructor(ngControlService) {
        this.ngControlService = ngControlService;
        this.subscriptions = [];
        // Implement our own status changes observable, since Angular controls don't
        this._statusChanges = new BehaviorSubject(CONTROL_STATE.NONE);
        // Wait for the control to be available
        this.subscriptions.push(this.ngControlService.controlChanges.subscribe(control => {
            if (control) {
                this.control = control;
                // Subscribe to the status change events, only after touched
                // and emit the control
                this.subscriptions.push(this.control.statusChanges.subscribe(() => {
                    this.triggerStatusChange();
                }));
            }
        }));
    }
    get statusChanges() {
        return this._statusChanges.asObservable();
    }
    triggerStatusChange() {
        /* Check if control is defined and run the code only then */
        if (this.control) {
            // These status values are mutually exclusive, so a control
            // cannot be both valid AND invalid or invalid AND disabled.
            const status = CONTROL_STATE[this.control.status];
            // At some point, with Angular 14, we started receiving the update for the control.touched
            // later than this moment in code. It depends on the order of imports of ClarityModule and
            // the FormsModule. This delay makes sure the touched state is properly updated.
            setTimeout(() => {
                this._statusChanges.next(this.control.touched && ['VALID', 'INVALID'].includes(status) ? status : CONTROL_STATE.NONE);
            });
        }
    }
    // Clean up subscriptions
    ngOnDestroy() {
        this.subscriptions.forEach(subscription => subscription.unsubscribe());
    }
}
IfControlStateService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: IfControlStateService, deps: [{ token: i1.NgControlService }], target: i0.ɵɵFactoryTarget.Injectable });
IfControlStateService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: IfControlStateService });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: IfControlStateService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.NgControlService }]; } });
