// @clr/angular/esm2020/layout/nav/aria-current-link.mjs
/*
 * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Directive } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
export class ClrAriaCurrentLink {
    constructor(rla) {
        this.rla = rla;
    }
    ngOnInit() {
        this.subscription = this.rla.isActiveChange.subscribe(isActive => {
            this.ariaCurrent = isActive ? 'page' : undefined;
        });
    }
    ngOnDestroy() {
        this.subscription.unsubscribe();
    }
}
ClrAriaCurrentLink.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: ClrAriaCurrentLink, deps: [{ token: i1.RouterLinkActive }], target: i0.ɵɵFactoryTarget.Directive });
ClrAriaCurrentLink.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "12.0.0", version: "13.3.11", type: ClrAriaCurrentLink, selector: "[clrAriaCurrentLink]", host: { properties: { "attr.aria-current": "ariaCurrent" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: ClrAriaCurrentLink, decorators: [{
            type: Directive,
            args: [{
                    selector: '[clrAriaCurrentLink]',
                    host: { '[attr.aria-current]': 'ariaCurrent' },
                }]
        }], ctorParameters: function () { return [{ type: i1.RouterLinkActive }]; } });
