// @clr/angular/esm2020/popover/tooltip/tooltip.mjs
/*
 * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Component, ElementRef } from '@angular/core';
import { ClrPopoverToggleService } from '../../utils/popover/providers/popover-toggle.service';
import { POPOVER_HOST_ANCHOR } from '../common/popover-host-anchor.token';
import { TooltipIdService } from './providers/tooltip-id.service';
import { TooltipMouseService } from './providers/tooltip-mouse.service';
import * as i0 from "@angular/core";
export class ClrTooltip {
}
ClrTooltip.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: ClrTooltip, deps: [], target: i0.ɵɵFactoryTarget.Component });
ClrTooltip.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "13.3.11", type: ClrTooltip, selector: "clr-tooltip", host: { properties: { "class.tooltip": "true" } }, providers: [
        ClrPopoverToggleService,
        { provide: POPOVER_HOST_ANCHOR, useExisting: ElementRef },
        TooltipIdService,
        TooltipMouseService,
    ], ngImport: i0, template: `<ng-content></ng-content>`, isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: ClrTooltip, decorators: [{
            type: Component,
            args: [{
                    selector: 'clr-tooltip',
                    template: `<ng-content></ng-content>`,
                    host: {
                        '[class.tooltip]': 'true',
                    },
                    providers: [
                        ClrPopoverToggleService,
                        { provide: POPOVER_HOST_ANCHOR, useExisting: ElementRef },
                        TooltipIdService,
                        TooltipMouseService,
                    ],
                }]
        }] });
