// @clr/angular/esm2020/utils/outside-click/outside-click.module.mjs
/*
 * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { OUSTIDE_CLICK_DIRECTIVES } from './index';
import * as i0 from "@angular/core";
import * as i1 from "./outside-click";
export class ClrOutsideClickModule {
}
ClrOutsideClickModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: ClrOutsideClickModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ClrOutsideClickModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: ClrOutsideClickModule, declarations: [i1.OutsideClick], imports: [CommonModule], exports: [i1.OutsideClick] });
ClrOutsideClickModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: ClrOutsideClickModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: ClrOutsideClickModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [OUSTIDE_CLICK_DIRECTIVES],
                    exports: [OUSTIDE_CLICK_DIRECTIVES],
                }]
        }] });
