// @cds/core/tokens/tokens.js
// This token format is currently experimental and may change in the future
export const globalBase = 'var(--cds-global-base)';
export const globalScaleSpace = 'var(--cds-global-scale-space)';
export const globalScaleLayoutSpace = 'var(--cds-global-scale-layout-space)';
export const globalScaleTypography = 'var(--cds-global-scale-typography)';
export const globalLayoutGridCols = 'var(--cds-global-layout-grid-cols)';
export const globalLayoutWidthXs = 'var(--cds-global-layout-width-xs)';
export const globalLayoutWidthSm = 'var(--cds-global-layout-width-sm)';
export const globalLayoutWidthMd = 'var(--cds-global-layout-width-md)';
export const globalLayoutWidthLg = 'var(--cds-global-layout-width-lg)';
export const globalLayoutWidthXl = 'var(--cds-global-layout-width-xl)';
export const globalLayoutSpaceXxxs = 'var(--cds-global-layout-space-xxxs)';
export const globalLayoutSpaceXxs = 'var(--cds-global-layout-space-xxs)';
export const globalLayoutSpaceXs = 'var(--cds-global-layout-space-xs)';
export const globalLayoutSpaceSm = 'var(--cds-global-layout-space-sm)';
export const globalLayoutSpaceMd = 'var(--cds-global-layout-space-md)';
export const globalLayoutSpaceLg = 'var(--cds-global-layout-space-lg)';
export const globalLayoutSpaceXl = 'var(--cds-global-layout-space-xl)';
export const globalLayoutSpaceXxl = 'var(--cds-global-layout-space-xxl)';
export const globalLayoutSpaceXxxl = 'var(--cds-global-layout-space-xxxl)';
export const globalSpace0 = 'var(--cds-global-space-0)';
export const globalSpace1 = 'var(--cds-global-space-1)';
export const globalSpace2 = 'var(--cds-global-space-2)';
export const globalSpace3 = 'var(--cds-global-space-3)';
export const globalSpace4 = 'var(--cds-global-space-4)';
export const globalSpace5 = 'var(--cds-global-space-5)';
export const globalSpace6 = 'var(--cds-global-space-6)';
export const globalSpace7 = 'var(--cds-global-space-7)';
export const globalSpace8 = 'var(--cds-global-space-8)';
export const globalSpace9 = 'var(--cds-global-space-9)';
export const globalSpace10 = 'var(--cds-global-space-10)';
export const globalSpace11 = 'var(--cds-global-space-11)';
export const globalSpace12 = 'var(--cds-global-space-12)';
export const globalSpace13 = 'var(--cds-global-space-13)';
export const globalSpace14 = 'var(--cds-global-space-14)';
export const globalSpace15 = 'var(--cds-global-space-15)';
export const globalColorBlack = 'var(--cds-global-color-black)';
export const globalColorWhite = 'var(--cds-global-color-white)';
export const globalColorGreen50 = 'var(--cds-global-color-green-50)';
export const globalColorGreen100 = 'var(--cds-global-color-green-100)';
export const globalColorGreen200 = 'var(--cds-global-color-green-200)';
export const globalColorGreen300 = 'var(--cds-global-color-green-300)';
export const globalColorGreen400 = 'var(--cds-global-color-green-400)';
export const globalColorGreen500 = 'var(--cds-global-color-green-500)';
export const globalColorGreen600 = 'var(--cds-global-color-green-600)';
export const globalColorGreen700 = 'var(--cds-global-color-green-700)';
export const globalColorGreen800 = 'var(--cds-global-color-green-800)';
export const globalColorGreen900 = 'var(--cds-global-color-green-900)';
export const globalColorGreen1000 = 'var(--cds-global-color-green-1000)';
export const globalColorBlue50 = 'var(--cds-global-color-blue-50)';
export const globalColorBlue100 = 'var(--cds-global-color-blue-100)';
export const globalColorBlue200 = 'var(--cds-global-color-blue-200)';
export const globalColorBlue300 = 'var(--cds-global-color-blue-300)';
export const globalColorBlue400 = 'var(--cds-global-color-blue-400)';
export const globalColorBlue500 = 'var(--cds-global-color-blue-500)';
export const globalColorBlue600 = 'var(--cds-global-color-blue-600)';
export const globalColorBlue700 = 'var(--cds-global-color-blue-700)';
export const globalColorBlue800 = 'var(--cds-global-color-blue-800)';
export const globalColorBlue900 = 'var(--cds-global-color-blue-900)';
export const globalColorBlue1000 = 'var(--cds-global-color-blue-1000)';
export const globalColorViolet50 = 'var(--cds-global-color-violet-50)';
export const globalColorViolet100 = 'var(--cds-global-color-violet-100)';
export const globalColorViolet200 = 'var(--cds-global-color-violet-200)';
export const globalColorViolet300 = 'var(--cds-global-color-violet-300)';
export const globalColorViolet400 = 'var(--cds-global-color-violet-400)';
export const globalColorViolet500 = 'var(--cds-global-color-violet-500)';
export const globalColorViolet600 = 'var(--cds-global-color-violet-600)';
export const globalColorViolet700 = 'var(--cds-global-color-violet-700)';
export const globalColorViolet800 = 'var(--cds-global-color-violet-800)';
export const globalColorViolet900 = 'var(--cds-global-color-violet-900)';
export const globalColorViolet1000 = 'var(--cds-global-color-violet-1000)';
export const globalColorRed50 = 'var(--cds-global-color-red-50)';
export const globalColorRed100 = 'var(--cds-global-color-red-100)';
export const globalColorRed200 = 'var(--cds-global-color-red-200)';
export const globalColorRed300 = 'var(--cds-global-color-red-300)';
export const globalColorRed400 = 'var(--cds-global-color-red-400)';
export const globalColorRed500 = 'var(--cds-global-color-red-500)';
export const globalColorRed600 = 'var(--cds-global-color-red-600)';
export const globalColorRed700 = 'var(--cds-global-color-red-700)';
export const globalColorRed800 = 'var(--cds-global-color-red-800)';
export const globalColorRed900 = 'var(--cds-global-color-red-900)';
export const globalColorRed1000 = 'var(--cds-global-color-red-1000)';
export const globalColorOchre50 = 'var(--cds-global-color-ochre-50)';
export const globalColorOchre100 = 'var(--cds-global-color-ochre-100)';
export const globalColorOchre200 = 'var(--cds-global-color-ochre-200)';
export const globalColorOchre300 = 'var(--cds-global-color-ochre-300)';
export const globalColorOchre400 = 'var(--cds-global-color-ochre-400)';
export const globalColorOchre500 = 'var(--cds-global-color-ochre-500)';
export const globalColorOchre600 = 'var(--cds-global-color-ochre-600)';
export const globalColorOchre700 = 'var(--cds-global-color-ochre-700)';
export const globalColorOchre800 = 'var(--cds-global-color-ochre-800)';
export const globalColorOchre900 = 'var(--cds-global-color-ochre-900)';
export const globalColorOchre1000 = 'var(--cds-global-color-ochre-1000)';
export const globalColorLavender50 = 'var(--cds-global-color-lavender-50)';
export const globalColorLavender100 = 'var(--cds-global-color-lavender-100)';
export const globalColorLavender200 = 'var(--cds-global-color-lavender-200)';
export const globalColorLavender300 = 'var(--cds-global-color-lavender-300)';
export const globalColorLavender400 = 'var(--cds-global-color-lavender-400)';
export const globalColorLavender500 = 'var(--cds-global-color-lavender-500)';
export const globalColorLavender600 = 'var(--cds-global-color-lavender-600)';
export const globalColorLavender700 = 'var(--cds-global-color-lavender-700)';
export const globalColorLavender800 = 'var(--cds-global-color-lavender-800)';
export const globalColorLavender900 = 'var(--cds-global-color-lavender-900)';
export const globalColorLavender1000 = 'var(--cds-global-color-lavender-1000)';
export const globalColorAzure50 = 'var(--cds-global-color-azure-50)';
export const globalColorAzure100 = 'var(--cds-global-color-azure-100)';
export const globalColorAzure200 = 'var(--cds-global-color-azure-200)';
export const globalColorAzure300 = 'var(--cds-global-color-azure-300)';
export const globalColorAzure400 = 'var(--cds-global-color-azure-400)';
export const globalColorAzure500 = 'var(--cds-global-color-azure-500)';
export const globalColorAzure600 = 'var(--cds-global-color-azure-600)';
export const globalColorAzure700 = 'var(--cds-global-color-azure-700)';
export const globalColorAzure800 = 'var(--cds-global-color-azure-800)';
export const globalColorAzure900 = 'var(--cds-global-color-azure-900)';
export const globalColorAzure1000 = 'var(--cds-global-color-azure-1000)';
export const globalColorAqua50 = 'var(--cds-global-color-aqua-50)';
export const globalColorAqua100 = 'var(--cds-global-color-aqua-100)';
export const globalColorAqua200 = 'var(--cds-global-color-aqua-200)';
export const globalColorAqua300 = 'var(--cds-global-color-aqua-300)';
export const globalColorAqua400 = 'var(--cds-global-color-aqua-400)';
export const globalColorAqua500 = 'var(--cds-global-color-aqua-500)';
export const globalColorAqua600 = 'var(--cds-global-color-aqua-600)';
export const globalColorAqua700 = 'var(--cds-global-color-aqua-700)';
export const globalColorAqua800 = 'var(--cds-global-color-aqua-800)';
export const globalColorAqua900 = 'var(--cds-global-color-aqua-900)';
export const globalColorAqua1000 = 'var(--cds-global-color-aqua-1000)';
export const globalColorJade50 = 'var(--cds-global-color-jade-50)';
export const globalColorJade100 = 'var(--cds-global-color-jade-100)';
export const globalColorJade200 = 'var(--cds-global-color-jade-200)';
export const globalColorJade300 = 'var(--cds-global-color-jade-300)';
export const globalColorJade400 = 'var(--cds-global-color-jade-400)';
export const globalColorJade500 = 'var(--cds-global-color-jade-500)';
export const globalColorJade600 = 'var(--cds-global-color-jade-600)';
export const globalColorJade700 = 'var(--cds-global-color-jade-700)';
export const globalColorJade800 = 'var(--cds-global-color-jade-800)';
export const globalColorJade900 = 'var(--cds-global-color-jade-900)';
export const globalColorJade1000 = 'var(--cds-global-color-jade-1000)';
export const globalColorYellow50 = 'var(--cds-global-color-yellow-50)';
export const globalColorYellow100 = 'var(--cds-global-color-yellow-100)';
export const globalColorYellow200 = 'var(--cds-global-color-yellow-200)';
export const globalColorYellow300 = 'var(--cds-global-color-yellow-300)';
export const globalColorYellow400 = 'var(--cds-global-color-yellow-400)';
export const globalColorYellow500 = 'var(--cds-global-color-yellow-500)';
export const globalColorYellow600 = 'var(--cds-global-color-yellow-600)';
export const globalColorYellow700 = 'var(--cds-global-color-yellow-700)';
export const globalColorYellow800 = 'var(--cds-global-color-yellow-800)';
export const globalColorYellow900 = 'var(--cds-global-color-yellow-900)';
export const globalColorYellow1000 = 'var(--cds-global-color-yellow-1000)';
export const globalColorLime50 = 'var(--cds-global-color-lime-50)';
export const globalColorLime100 = 'var(--cds-global-color-lime-100)';
export const globalColorLime200 = 'var(--cds-global-color-lime-200)';
export const globalColorLime300 = 'var(--cds-global-color-lime-300)';
export const globalColorLime400 = 'var(--cds-global-color-lime-400)';
export const globalColorLime500 = 'var(--cds-global-color-lime-500)';
export const globalColorLime600 = 'var(--cds-global-color-lime-600)';
export const globalColorLime700 = 'var(--cds-global-color-lime-700)';
export const globalColorLime800 = 'var(--cds-global-color-lime-800)';
export const globalColorLime900 = 'var(--cds-global-color-lime-900)';
export const globalColorLime1000 = 'var(--cds-global-color-lime-1000)';
export const globalColorTangerine50 = 'var(--cds-global-color-tangerine-50)';
export const globalColorTangerine100 = 'var(--cds-global-color-tangerine-100)';
export const globalColorTangerine200 = 'var(--cds-global-color-tangerine-200)';
export const globalColorTangerine300 = 'var(--cds-global-color-tangerine-300)';
export const globalColorTangerine400 = 'var(--cds-global-color-tangerine-400)';
export const globalColorTangerine500 = 'var(--cds-global-color-tangerine-500)';
export const globalColorTangerine600 = 'var(--cds-global-color-tangerine-600)';
export const globalColorTangerine700 = 'var(--cds-global-color-tangerine-700)';
export const globalColorTangerine800 = 'var(--cds-global-color-tangerine-800)';
export const globalColorTangerine900 = 'var(--cds-global-color-tangerine-900)';
export const globalColorTangerine1000 = 'var(--cds-global-color-tangerine-1000)';
export const globalColorMagenta50 = 'var(--cds-global-color-magenta-50)';
export const globalColorMagenta100 = 'var(--cds-global-color-magenta-100)';
export const globalColorMagenta200 = 'var(--cds-global-color-magenta-200)';
export const globalColorMagenta300 = 'var(--cds-global-color-magenta-300)';
export const globalColorMagenta400 = 'var(--cds-global-color-magenta-400)';
export const globalColorMagenta500 = 'var(--cds-global-color-magenta-500)';
export const globalColorMagenta600 = 'var(--cds-global-color-magenta-600)';
export const globalColorMagenta700 = 'var(--cds-global-color-magenta-700)';
export const globalColorMagenta800 = 'var(--cds-global-color-magenta-800)';
export const globalColorMagenta900 = 'var(--cds-global-color-magenta-900)';
export const globalColorMagenta1000 = 'var(--cds-global-color-magenta-1000)';
export const globalColorPink50 = 'var(--cds-global-color-pink-50)';
export const globalColorPink100 = 'var(--cds-global-color-pink-100)';
export const globalColorPink200 = 'var(--cds-global-color-pink-200)';
export const globalColorPink300 = 'var(--cds-global-color-pink-300)';
export const globalColorPink400 = 'var(--cds-global-color-pink-400)';
export const globalColorPink500 = 'var(--cds-global-color-pink-500)';
export const globalColorPink600 = 'var(--cds-global-color-pink-600)';
export const globalColorPink700 = 'var(--cds-global-color-pink-700)';
export const globalColorPink800 = 'var(--cds-global-color-pink-800)';
export const globalColorPink900 = 'var(--cds-global-color-pink-900)';
export const globalColorPink1000 = 'var(--cds-global-color-pink-1000)';
export const globalColorWarmGray50 = 'var(--cds-global-color-warm-gray-50)';
export const globalColorWarmGray100 = 'var(--cds-global-color-warm-gray-100)';
export const globalColorWarmGray200 = 'var(--cds-global-color-warm-gray-200)';
export const globalColorWarmGray300 = 'var(--cds-global-color-warm-gray-300)';
export const globalColorWarmGray400 = 'var(--cds-global-color-warm-gray-400)';
export const globalColorWarmGray500 = 'var(--cds-global-color-warm-gray-500)';
export const globalColorWarmGray600 = 'var(--cds-global-color-warm-gray-600)';
export const globalColorWarmGray700 = 'var(--cds-global-color-warm-gray-700)';
export const globalColorWarmGray800 = 'var(--cds-global-color-warm-gray-800)';
export const globalColorWarmGray900 = 'var(--cds-global-color-warm-gray-900)';
export const globalColorWarmGray1000 = 'var(--cds-global-color-warm-gray-1000)';
export const globalColorSlate50 = 'var(--cds-global-color-slate-50)';
export const globalColorSlate100 = 'var(--cds-global-color-slate-100)';
export const globalColorSlate200 = 'var(--cds-global-color-slate-200)';
export const globalColorSlate300 = 'var(--cds-global-color-slate-300)';
export const globalColorSlate400 = 'var(--cds-global-color-slate-400)';
export const globalColorSlate500 = 'var(--cds-global-color-slate-500)';
export const globalColorSlate600 = 'var(--cds-global-color-slate-600)';
export const globalColorSlate700 = 'var(--cds-global-color-slate-700)';
export const globalColorSlate800 = 'var(--cds-global-color-slate-800)';
export const globalColorSlate900 = 'var(--cds-global-color-slate-900)';
export const globalColorSlate1000 = 'var(--cds-global-color-slate-1000)';
export const globalColorIce50 = 'var(--cds-global-color-ice-50)';
export const globalColorIce100 = 'var(--cds-global-color-ice-100)';
export const globalColorIce200 = 'var(--cds-global-color-ice-200)';
export const globalColorIce300 = 'var(--cds-global-color-ice-300)';
export const globalColorIce400 = 'var(--cds-global-color-ice-400)';
export const globalColorIce500 = 'var(--cds-global-color-ice-500)';
export const globalColorIce600 = 'var(--cds-global-color-ice-600)';
export const globalColorIce700 = 'var(--cds-global-color-ice-700)';
export const globalColorIce800 = 'var(--cds-global-color-ice-800)';
export const globalColorIce900 = 'var(--cds-global-color-ice-900)';
export const globalColorIce1000 = 'var(--cds-global-color-ice-1000)';
export const globalColorCoolGray50 = 'var(--cds-global-color-cool-gray-50)';
export const globalColorCoolGray100 = 'var(--cds-global-color-cool-gray-100)';
export const globalColorCoolGray200 = 'var(--cds-global-color-cool-gray-200)';
export const globalColorCoolGray300 = 'var(--cds-global-color-cool-gray-300)';
export const globalColorCoolGray400 = 'var(--cds-global-color-cool-gray-400)';
export const globalColorCoolGray500 = 'var(--cds-global-color-cool-gray-500)';
export const globalColorCoolGray600 = 'var(--cds-global-color-cool-gray-600)';
export const globalColorCoolGray700 = 'var(--cds-global-color-cool-gray-700)';
export const globalColorCoolGray800 = 'var(--cds-global-color-cool-gray-800)';
export const globalColorCoolGray900 = 'var(--cds-global-color-cool-gray-900)';
export const globalColorCoolGray1000 = 'var(--cds-global-color-cool-gray-1000)';
export const globalColorTan50 = 'var(--cds-global-color-tan-50)';
export const globalColorTan100 = 'var(--cds-global-color-tan-100)';
export const globalColorTan200 = 'var(--cds-global-color-tan-200)';
export const globalColorTan300 = 'var(--cds-global-color-tan-300)';
export const globalColorTan400 = 'var(--cds-global-color-tan-400)';
export const globalColorTan500 = 'var(--cds-global-color-tan-500)';
export const globalColorTan600 = 'var(--cds-global-color-tan-600)';
export const globalColorTan700 = 'var(--cds-global-color-tan-700)';
export const globalColorTan800 = 'var(--cds-global-color-tan-800)';
export const globalColorTan900 = 'var(--cds-global-color-tan-900)';
export const globalColorTan1000 = 'var(--cds-global-color-tan-1000)';
export const globalColorConstruction50 = 'var(--cds-global-color-construction-50)';
export const globalColorConstruction100 = 'var(--cds-global-color-construction-100)';
export const globalColorConstruction200 = 'var(--cds-global-color-construction-200)';
export const globalColorConstruction300 = 'var(--cds-global-color-construction-300)';
export const globalColorConstruction400 = 'var(--cds-global-color-construction-400)';
export const globalColorConstruction500 = 'var(--cds-global-color-construction-500)';
export const globalColorConstruction600 = 'var(--cds-global-color-construction-600)';
export const globalColorConstruction700 = 'var(--cds-global-color-construction-700)';
export const globalColorConstruction800 = 'var(--cds-global-color-construction-800)';
export const globalColorConstruction900 = 'var(--cds-global-color-construction-900)';
export const globalColorConstruction1000 = 'var(--cds-global-color-construction-1000)';
export const globalColorGray0 = 'var(--cds-global-color-gray-0)';
export const globalColorGray50 = 'var(--cds-global-color-gray-50)';
export const globalColorGray100 = 'var(--cds-global-color-gray-100)';
export const globalColorGray200 = 'var(--cds-global-color-gray-200)';
export const globalColorGray300 = 'var(--cds-global-color-gray-300)';
export const globalColorGray400 = 'var(--cds-global-color-gray-400)';
export const globalColorGray500 = 'var(--cds-global-color-gray-500)';
export const globalColorGray600 = 'var(--cds-global-color-gray-600)';
export const globalColorGray700 = 'var(--cds-global-color-gray-700)';
export const globalColorGray800 = 'var(--cds-global-color-gray-800)';
export const globalColorGray900 = 'var(--cds-global-color-gray-900)';
export const globalColorGray1000 = 'var(--cds-global-color-gray-1000)';
export const globalTypographyColor100 = 'var(--cds-global-typography-color-100)';
export const globalTypographyColor200 = 'var(--cds-global-typography-color-200)';
export const globalTypographyColor300 = 'var(--cds-global-typography-color-300)';
export const globalTypographyColor400 = 'var(--cds-global-typography-color-400)';
export const globalTypographyColor500 = 'var(--cds-global-typography-color-500)';
export const globalTypographyFontWeightLight = 'var(--cds-global-typography-font-weight-light)';
export const globalTypographyFontWeightRegular = 'var(--cds-global-typography-font-weight-regular)';
export const globalTypographyFontWeightMedium = 'var(--cds-global-typography-font-weight-medium)';
export const globalTypographyFontWeightSemibold = 'var(--cds-global-typography-font-weight-semibold)';
export const globalTypographyFontWeightBold = 'var(--cds-global-typography-font-weight-bold)';
export const globalTypographyFontWeightExtrabold = 'var(--cds-global-typography-font-weight-extrabold)';
export const globalTypographyFontSize0 = 'var(--cds-global-typography-font-size-0)';
export const globalTypographyFontSize1 = 'var(--cds-global-typography-font-size-1)';
export const globalTypographyFontSize2 = 'var(--cds-global-typography-font-size-2)';
export const globalTypographyFontSize3 = 'var(--cds-global-typography-font-size-3)';
export const globalTypographyFontSize4 = 'var(--cds-global-typography-font-size-4)';
export const globalTypographyFontSize5 = 'var(--cds-global-typography-font-size-5)';
export const globalTypographyFontSize6 = 'var(--cds-global-typography-font-size-6)';
export const globalTypographyFontSize7 = 'var(--cds-global-typography-font-size-7)';
export const globalTypographyFontSize8 = 'var(--cds-global-typography-font-size-8)';
export const globalTypographyFontSize9 = 'var(--cds-global-typography-font-size-9)';
export const globalTypographyFontFamily = 'var(--cds-global-typography-font-family)';
export const globalTypographyHeaderFontFamily = 'var(--cds-global-typography-header-font-family)';
export const globalTypographyMonospaceFontFamily = 'var(--cds-global-typography-monospace-font-family)';
export const globalTypographyTopGapHeight = 'var(--cds-global-typography-top-gap-height)';
export const globalTypographyAscenderHeight = 'var(--cds-global-typography-ascender-height)';
export const globalTypographyXHeight = 'var(--cds-global-typography-x-height)';
export const globalTypographyLinkColor = 'var(--cds-global-typography-link-color)';
export const globalTypographyLinkColorHover = 'var(--cds-global-typography-link-color-hover)';
export const globalTypographyLinkColorVisited = 'var(--cds-global-typography-link-color-visited)';
export const globalTypographyLinkColorVisitedHover = 'var(--cds-global-typography-link-color-visited-hover)';
export const globalTypographyBodyFontSize = 'var(--cds-global-typography-body-font-size)';
export const globalTypographyBodyLineHeight = 'var(--cds-global-typography-body-line-height)';
export const globalTypographyBodyLetterSpacing = 'var(--cds-global-typography-body-letter-spacing)';
export const globalTypographyBodyFontWeight = 'var(--cds-global-typography-body-font-weight)';
export const globalTypographyDisplayFontSize = 'var(--cds-global-typography-display-font-size)';
export const globalTypographyDisplayLineHeight = 'var(--cds-global-typography-display-line-height)';
export const globalTypographyDisplayLetterSpacing = 'var(--cds-global-typography-display-letter-spacing)';
export const globalTypographyDisplayFontWeight = 'var(--cds-global-typography-display-font-weight)';
export const globalTypographyHeadingFontSize = 'var(--cds-global-typography-heading-font-size)';
export const globalTypographyHeadingLineHeight = 'var(--cds-global-typography-heading-line-height)';
export const globalTypographyHeadingLetterSpacing = 'var(--cds-global-typography-heading-letter-spacing)';
export const globalTypographyHeadingFontWeight = 'var(--cds-global-typography-heading-font-weight)';
export const globalTypographyTitleFontSize = 'var(--cds-global-typography-title-font-size)';
export const globalTypographyTitleLineHeight = 'var(--cds-global-typography-title-line-height)';
export const globalTypographyTitleLetterSpacing = 'var(--cds-global-typography-title-letter-spacing)';
export const globalTypographyTitleFontWeight = 'var(--cds-global-typography-title-font-weight)';
export const globalTypographySectionFontSize = 'var(--cds-global-typography-section-font-size)';
export const globalTypographySectionLineHeight = 'var(--cds-global-typography-section-line-height)';
export const globalTypographySectionLetterSpacing = 'var(--cds-global-typography-section-letter-spacing)';
export const globalTypographySectionFontWeight = 'var(--cds-global-typography-section-font-weight)';
export const globalTypographySubsectionFontSize = 'var(--cds-global-typography-subsection-font-size)';
export const globalTypographySubsectionLineHeight = 'var(--cds-global-typography-subsection-line-height)';
export const globalTypographySubsectionLetterSpacing = 'var(--cds-global-typography-subsection-letter-spacing)';
export const globalTypographySubsectionFontWeight = 'var(--cds-global-typography-subsection-font-weight)';
export const globalTypographyMessageFontSize = 'var(--cds-global-typography-message-font-size)';
export const globalTypographyMessageLineHeight = 'var(--cds-global-typography-message-line-height)';
export const globalTypographyMessageLetterSpacing = 'var(--cds-global-typography-message-letter-spacing)';
export const globalTypographyMessageFontWeight = 'var(--cds-global-typography-message-font-weight)';
export const globalTypographySecondaryFontSize = 'var(--cds-global-typography-secondary-font-size)';
export const globalTypographySecondaryLineHeight = 'var(--cds-global-typography-secondary-line-height)';
export const globalTypographySecondaryLetterSpacing = 'var(--cds-global-typography-secondary-letter-spacing)';
export const globalTypographySecondaryFontWeight = 'var(--cds-global-typography-secondary-font-weight)';
export const globalTypographyCaptionFontSize = 'var(--cds-global-typography-caption-font-size)';
export const globalTypographyCaptionLineHeight = 'var(--cds-global-typography-caption-line-height)';
export const globalTypographyCaptionLetterSpacing = 'var(--cds-global-typography-caption-letter-spacing)';
export const globalTypographyCaptionFontWeight = 'var(--cds-global-typography-caption-font-weight)';
export const globalTypographySmallcaptionFontSize = 'var(--cds-global-typography-smallcaption-font-size)';
export const globalTypographySmallcaptionLineHeight = 'var(--cds-global-typography-smallcaption-line-height)';
export const globalTypographySmallcaptionLetterSpacing = 'var(--cds-global-typography-smallcaption-letter-spacing)';
export const globalTypographySmallcaptionFontWeight = 'var(--cds-global-typography-smallcaption-font-weight)';
export const globalAnimationDurationInstant = 'var(--cds-global-animation-duration-instant)';
export const globalAnimationDurationQuickest = 'var(--cds-global-animation-duration-quickest)';
export const globalAnimationDurationQuicker = 'var(--cds-global-animation-duration-quicker)';
export const globalAnimationDurationQuick = 'var(--cds-global-animation-duration-quick)';
export const globalAnimationDurationSecondary = 'var(--cds-global-animation-duration-secondary)';
export const globalAnimationDurationPrimary = 'var(--cds-global-animation-duration-primary)';
export const globalAnimationDurationSlow = 'var(--cds-global-animation-duration-slow)';
export const globalAnimationDurationSlower = 'var(--cds-global-animation-duration-slower)';
export const globalAnimationDurationSlowest = 'var(--cds-global-animation-duration-slowest)';
export const globalAnimationEasingPrimary = 'var(--cds-global-animation-easing-primary)';
export const globalAnimationEasingSecondary = 'var(--cds-global-animation-easing-secondary)';
export const globalAnimationEasingLoop = 'var(--cds-global-animation-easing-loop)';
export const aliasObjectBorderRadius100 = 'var(--cds-alias-object-border-radius-100)';
export const aliasObjectBorderRadius200 = 'var(--cds-alias-object-border-radius-200)';
export const aliasObjectBorderRadius300 = 'var(--cds-alias-object-border-radius-300)';
export const aliasObjectBorderWidth100 = 'var(--cds-alias-object-border-width-100)';
export const aliasObjectBorderWidth200 = 'var(--cds-alias-object-border-width-200)';
export const aliasObjectBorderWidth300 = 'var(--cds-alias-object-border-width-300)';
export const aliasObjectBorderWidth400 = 'var(--cds-alias-object-border-width-400)';
export const aliasObjectBorderColor = 'var(--cds-alias-object-border-color)';
export const aliasObjectBorderColorTint = 'var(--cds-alias-object-border-color-tint)';
export const aliasObjectBorderColorShade = 'var(--cds-alias-object-border-color-shade)';
export const aliasObjectShadow100 = 'var(--cds-alias-object-shadow-100)';
export const aliasObjectShadow200 = 'var(--cds-alias-object-shadow-200)';
export const aliasObjectShadow300 = 'var(--cds-alias-object-shadow-300)';
export const aliasObjectOpacity0 = 'var(--cds-alias-object-opacity-0)';
export const aliasObjectOpacity100 = 'var(--cds-alias-object-opacity-100)';
export const aliasObjectOpacity200 = 'var(--cds-alias-object-opacity-200)';
export const aliasObjectOpacity300 = 'var(--cds-alias-object-opacity-300)';
export const aliasObjectInteractionOutline = 'var(--cds-alias-object-interaction-outline)';
export const aliasObjectInteractionOutlineWebkit = 'var(--cds-alias-object-interaction-outline-webkit)';
export const aliasObjectInteractionOutlineOffset = 'var(--cds-alias-object-interaction-outline-offset)';
export const aliasObjectInteractionTouchTarget = 'var(--cds-alias-object-interaction-touch-target)';
export const aliasObjectInteractionBorderColor = 'var(--cds-alias-object-interaction-border-color)';
export const aliasObjectInteractionBackground = 'var(--cds-alias-object-interaction-background)';
export const aliasObjectInteractionBackgroundHover = 'var(--cds-alias-object-interaction-background-hover)';
export const aliasObjectInteractionBackgroundActive = 'var(--cds-alias-object-interaction-background-active)';
export const aliasObjectInteractionBackgroundSelected = 'var(--cds-alias-object-interaction-background-selected)';
export const aliasObjectInteractionBackgroundDisabled = 'var(--cds-alias-object-interaction-background-disabled)';
export const aliasObjectInteractionBackgroundHighlight = 'var(--cds-alias-object-interaction-background-highlight)';
export const aliasObjectInteractionColor = 'var(--cds-alias-object-interaction-color)';
export const aliasObjectInteractionColorHover = 'var(--cds-alias-object-interaction-color-hover)';
export const aliasObjectInteractionColorActive = 'var(--cds-alias-object-interaction-color-active)';
export const aliasObjectInteractionColorSelected = 'var(--cds-alias-object-interaction-color-selected)';
export const aliasObjectInteractionColorDisabled = 'var(--cds-alias-object-interaction-color-disabled)';
export const aliasObjectAppBackground = 'var(--cds-alias-object-app-background)';
export const aliasObjectOverlayBackground = 'var(--cds-alias-object-overlay-background)';
export const aliasObjectOverlayBackdropBackground = 'var(--cds-alias-object-overlay-backdrop-background)';
export const aliasObjectContainerBackground = 'var(--cds-alias-object-container-background)';
export const aliasObjectContainerBackgroundTint = 'var(--cds-alias-object-container-background-tint)';
export const aliasObjectContainerBackgroundShade = 'var(--cds-alias-object-container-background-shade)';
export const aliasObjectContainerBorderColor = 'var(--cds-alias-object-container-border-color)';
export const aliasStatusInfo = 'var(--cds-alias-status-info)';
export const aliasStatusInfoTint = 'var(--cds-alias-status-info-tint)';
export const aliasStatusInfoShade = 'var(--cds-alias-status-info-shade)';
export const aliasStatusSuccess = 'var(--cds-alias-status-success)';
export const aliasStatusSuccessTint = 'var(--cds-alias-status-success-tint)';
export const aliasStatusSuccessShade = 'var(--cds-alias-status-success-shade)';
export const aliasStatusWarning = 'var(--cds-alias-status-warning)';
export const aliasStatusWarningTint = 'var(--cds-alias-status-warning-tint)';
export const aliasStatusWarningShade = 'var(--cds-alias-status-warning-shade)';
export const aliasStatusWarningDark = 'var(--cds-alias-status-warning-dark)';
export const aliasStatusDanger = 'var(--cds-alias-status-danger)';
export const aliasStatusDangerTint = 'var(--cds-alias-status-danger-tint)';
export const aliasStatusDangerShade = 'var(--cds-alias-status-danger-shade)';
export const aliasStatusDangerDark = 'var(--cds-alias-status-danger-dark)';
export const aliasStatusNeutral = 'var(--cds-alias-status-neutral)';
export const aliasStatusNeutralTint = 'var(--cds-alias-status-neutral-tint)';
export const aliasStatusNeutralShade = 'var(--cds-alias-status-neutral-shade)';
export const aliasStatusDisabled = 'var(--cds-alias-status-disabled)';
export const aliasStatusDisabledTint = 'var(--cds-alias-status-disabled-tint)';
export const aliasStatusDisabledShade = 'var(--cds-alias-status-disabled-shade)';
export const aliasStatusAlt = 'var(--cds-alias-status-alt)';
export const aliasStatusAltTint = 'var(--cds-alias-status-alt-tint)';
export const aliasStatusAltShade = 'var(--cds-alias-status-alt-shade)';
export const internalScale1 = 'var(--cds-internal-scale-1)';
export const internalScale2 = 'var(--cds-internal-scale-2)';
export const internalScale3 = 'var(--cds-internal-scale-3)';
