// @clr/angular/esm2020/utils/focus-trap/focus-trap-tracker.service.mjs
/*
 * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { Injectable } from '@angular/core';
import * as i0 from "@angular/core";
export class FocusTrapTracker {
    constructor() {
        this._previousFocusTraps = [];
    }
    get current() {
        return this._current;
    }
    set current(value) {
        this._previousFocusTraps.push(this._current);
        this._current = value;
    }
    get nbFocusTrappers() {
        return this._previousFocusTraps.length;
    }
    activatePreviousTrapper() {
        this._current = this._previousFocusTraps.pop();
    }
}
FocusTrapTracker.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: FocusTrapTracker, deps: [], target: i0.ɵɵFactoryTarget.Injectable });
FocusTrapTracker.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: FocusTrapTracker, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: FocusTrapTracker, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }] });
