// @clr/angular/esm2020/utils/focus-trap/focus-trap.mjs
/*
 * Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
 * This software is released under MIT license.
 * The full license information can be found in LICENSE in the root directory of this project.
 */
import { DOCUMENT, isPlatformBrowser } from '@angular/common';
import { Injectable } from '@angular/core';
import { createId, FocusTrapTrackerService, isFocusable, isHTMLElement } from '@cds/core/internal';
import * as i0 from "@angular/core";
export function refocusIfOutsideFocusTrapElement(focusedElement, focusTrapElement, elementToRefocus) {
    const focusTrapIsCurrent = FocusTrapTrackerService.getCurrent() === focusTrapElement;
    const elementToFocusIsOutsideFocusTrap = elementIsOutsideFocusTrapElement(focusedElement, focusTrapElement);
    if (focusTrapIsCurrent && elementToFocusIsOutsideFocusTrap) {
        elementToRefocus = elementToRefocus || focusTrapElement;
        elementToRefocus.focus();
    }
    else {
        focusedElement.focus();
    }
}
export function elementIsOutsideFocusTrapElement(focusedElement, focusTrapElement) {
    if (focusedElement === focusTrapElement.topReboundElement ||
        focusedElement === focusTrapElement.bottomReboundElement) {
        return true;
    }
    const elementIsInFocusTrapLightDom = focusTrapElement.contains(focusedElement);
    if (elementIsInFocusTrapLightDom) {
        return false;
    }
    if (focusTrapElement !== null && focusTrapElement.contains(focusedElement)) {
        return false;
    }
    return true;
}
export function createFocusTrapReboundElement(document) {
    const offScreenSpan = document.createElement('span');
    offScreenSpan.setAttribute('tabindex', '0');
    offScreenSpan.classList.add('offscreen-focus-rebounder');
    return offScreenSpan;
}
export function addReboundElementsToFocusTrapElement(document, focusTrapElement) {
    if (focusTrapElement && !focusTrapElement.topReboundElement && !focusTrapElement.bottomReboundElement) {
        focusTrapElement.topReboundElement = createFocusTrapReboundElement(document);
        focusTrapElement.bottomReboundElement = createFocusTrapReboundElement(document);
        const parent = focusTrapElement.parentElement;
        const sibling = focusTrapElement.nextSibling;
        if (parent) {
            parent.insertBefore(focusTrapElement.topReboundElement, focusTrapElement);
            if (sibling) {
                parent.insertBefore(focusTrapElement.bottomReboundElement, sibling);
            }
            else {
                parent.appendChild(focusTrapElement.bottomReboundElement);
            }
        }
    }
}
export function removeReboundElementsFromFocusTrapElement(focusTrapElement) {
    if (focusTrapElement) {
        const parent = focusTrapElement.parentElement;
        if (parent) {
            const topRebound = focusTrapElement.topReboundElement;
            const bottomRebound = focusTrapElement.bottomReboundElement;
            if (topRebound) {
                parent.removeChild(topRebound);
            }
            if (bottomRebound) {
                parent.removeChild(bottomRebound);
            }
        }
        // These are here to to make sure that we completely delete all traces of the removed DOM objects.
        delete focusTrapElement.topReboundElement;
        delete focusTrapElement.bottomReboundElement;
    }
}
// this helper exists to enable the focus trap class to handle vanilla html elements
// it's primary concern is to keep TS happy.
export function castHtmlElementToFocusTrapElement(el) {
    return el;
}
export class FocusTrap {
    constructor(renderer, injector, platformId, hostElement) {
        this.renderer = renderer;
        this.active = false;
        if (isPlatformBrowser(platformId)) {
            this._document = injector.get(DOCUMENT);
        }
        hostElement = castHtmlElementToFocusTrapElement(hostElement);
        if (!hostElement.focusTrapId) {
            hostElement.focusTrapId = createId();
        }
        this.focusTrapElement = hostElement;
    }
    enableFocusTrap() {
        const fte = this.focusTrapElement;
        const firstFocusElement = fte.querySelector('[cds-first-focus]');
        const activeEl = this._document.activeElement;
        if (FocusTrapTrackerService.getCurrent() === fte) {
            throw new Error('Focus trap is already enabled for this instance.');
        }
        this.firstFocusElement = firstFocusElement || this.focusTrapElement;
        addReboundElementsToFocusTrapElement(this._document, fte);
        if (!isFocusable(fte)) {
            fte.setAttribute('tabindex', '-1');
        }
        if (activeEl && isHTMLElement(activeEl)) {
            this.previousFocus = activeEl;
        }
        FocusTrapTrackerService.setCurrent(fte);
        // setTimeout here is required for Safari which may try to set focus on
        // element before it is visible...
        const focusTimer = setTimeout(() => {
            this.firstFocusElement.focus();
            clearTimeout(focusTimer);
        }, 10);
        this.onFocusInEvent = this.onFocusIn.bind(this);
        this.unlisten = this.renderer.listen(this._document, 'focusin', this.onFocusInEvent);
        this.active = true;
    }
    removeFocusTrap() {
        if (this.unlisten) {
            this.unlisten();
        }
        removeReboundElementsFromFocusTrapElement(this.focusTrapElement);
        this.renderer.removeAttribute(this.focusTrapElement, 'tabindex');
        FocusTrapTrackerService.activatePreviousCurrent();
        this.active = false;
        if (this.previousFocus) {
            this.previousFocus.focus();
        }
    }
    onFocusIn(event) {
        refocusIfOutsideFocusTrapElement(event.composedPath()[0], this.focusTrapElement, this.firstFocusElement);
    }
}
FocusTrap.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: FocusTrap, deps: "invalid", target: i0.ɵɵFactoryTarget.Injectable });
FocusTrap.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: FocusTrap });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.3.11", ngImport: i0, type: FocusTrap, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i0.Renderer2 }, { type: i0.Injector }, { type: undefined }, { type: undefined }]; } });
